addons/website_event/tests/test_website_event.py

```diff
@@ -58,6 +59,41 @@ def test_canonical_url(self):
 
 @tagged('post_install', '-at_install')
 class TestWebsiteAccess(HttpCaseWithUserDemo, TestWebsiteEventCommon):
+    @staticmethod
+    def are_names_in_order(names_in_expected_order, text):
+        last_index = -1
+        if isinstance(text, bytes):
+            text = text.decode('utf-8')
+        for name in names_in_expected_order:
+            current_index = text.find(name, last_index + 1)
+            if current_index == -1:
+                if text.find(name) != -1:
+                    return False
+                else:
+                    raise Exception(f"I can't check ordering when name: '{name}' is not present in the text.")
+            elif current_index <= last_index:
+                return False
+            last_index = current_index
+        return True
+
+    @freeze_time('2024-09-1')
+    def test_event_ordering(self):
+        all_events = self.env['event.event'].search([])
+        all_events.write({
+            'date_begin': datetime(2024, 10, 1, 10, 0, 0),
+            'date_end': datetime(2024, 10, 1, 14, 0, 0),
+            'is_published': True,
+            'website_published': True,
+            'website_id': 1
+        })
+        self.env['event.event'].flush()
+
+        res = self.url_open('/event?date=all')
+
+        all_events_sorted = all_events.sorted(key=lambda order: (order.id), reverse=True)
+        names_in_expected_order = all_events_sorted.mapped('name')[:12]
```

Comment:
12 should be justified, otherwise it just seems arbitrary

Limit should have been on the search, but if we use test data would say we can just remove the limit as we know there should only be 3